app/models/session.rb

```diff
@@ -209,8 +209,10 @@ def as_json(opts=nil)
       end
     end
 
-    first_measurement = fetch_first_measurement_coordinates
-    res.merge!(:startingLongitude => first_measurement.longitude, :startingLatitude => first_measurement.latitude)
+    if type == "MobileSession"
+      first_measurement = fetch_first_measurement_coordinates
+      res.merge!(:startingLongitude => first_measurement.longitude, :startingLatitude => first_measurement.latitude, :average => measurements.average(:value))
```

Comment:
I'd prefer to keep ruby in snake_case and js in camelCase. We should have another layer doing the translation

Reply:
I guess this should be optimized? Also this method is screaming to be refactored!